libminifi/src/utils/net/AsioSocketUtils.cpp

```diff
@@ -44,4 +47,66 @@ asio::ssl::context getSslContext(const controllers::SSLContextService& ssl_conte
   return ssl_context;
 }
 
+AsioSocketConnection::AsioSocketConnection(SocketData socket_data) : socket_data_(std::move(socket_data)) {
+}
+
+int AsioSocketConnection::initialize() {
+  bool result = false;
+  if (socket_data_.ssl_context_service) {
+    result = connectTcpSocketOverSsl();
+  } else {
+    result = connectTcpSocket();
+  }
+  return result ? 0 : -1;
+}
+
+bool AsioSocketConnection::connectTcpSocketOverSsl() {
+  auto ssl_context = utils::net::getSslContext(*socket_data_.ssl_context_service);
+  asio::ssl::stream<asio::ip::tcp::socket> socket(io_context_, ssl_context);
+
+  bindToLocalInterface(socket.lowest_layer());
```

Comment:
As mentioned in the previous comment: client sockets (that initiate a connection) should probably not bind to anything.

Reply:
Updated in a42e21d6c626774f5b9d38b021727a9b37c9eafe